### PR TITLE
feat(TDP-5218): Accept several authorities

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this Pull Request is trying to solve?**
Authority check can be performed only with one value.
If you want to check if a user has one of two authorities, you will be stuck. 

**What is the chosen solution to this problem?**
We can pass an array of authority now, and if one of them is owned by the user, then he can access the resource.
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5218
 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java [hunk 1]
```diff
@@ -67,10 +69,31 @@ public Object requires(ProceedingJoinPoint pjp) throws Throwable {
...
+
+        AtomicBoolean checkOk = new AtomicBoolean(false);
+
+        final String authority = annotation.authority();
+        final String[] values = annotation.values();
+        final Supplier<Stream<String>> streamSupplier = () -> Stream.of(values);
+        final String value = annotation.value();
+
+        if (StringUtils.isNotBlank(authority)) {
+            if (isAllowed(authority)) {
```
Comment (pbailly) [inline daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java]: You could avoid the if with:
`checkOk.set(isAllowed(authority));`

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java [hunk 2]
```diff
@@ -67,10 +69,31 @@ public Object requires(ProceedingJoinPoint pjp) throws Throwable {
...
+                checkOk.set(true);
+            }
+        } else if (streamSupplier.get().anyMatch(StringUtils::isNotBlank)) {
+            streamSupplier.get().filter(StringUtils::isNotBlank).forEach(currentAuthority -> {
+                if (isAllowed(currentAuthority)) {
```
Comment (pbailly) [inline daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java]: You could avoid the if with:
`checkOk.set(isAllowed(currentAuthority));`

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java [hunk 3]
```diff
@@ -67,10 +69,31 @@ public Object requires(ProceedingJoinPoint pjp) throws Throwable {
...
+                    checkOk.set(true);
+                }
+            });
+        } else if (StringUtils.isNotBlank(value)) {
+            if (isAllowed(value)) {
```
Comment (pbailly) [inline daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java]: You could avoid the if with:
`checkOk.set(isAllowed(value));`

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java (hunk 2 above)
Comment (frassinier) [reply daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java]: No, I can't. I need to change the flag only if it's truthy!

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/36/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/36/)

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java (hunk 2 above)
Comment (pbailly) [reply daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java]: And what if you do the if in another filter?

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/37/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/37/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/38/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/38/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/42/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/42/)

File: daikon-spring/daikon-spring-security/src/test/java/org/talend/daikon/security/access/RequiresTestComponent.java [hunk 4]
```diff
@@ -17,8 +17,29 @@
 @Component
 public class RequiresTestComponent {
```
Comment (sgandon) [inline daikon-spring/daikon-spring-security/src/test/java/org/talend/daikon/security/access/RequiresTestComponent.java]: There is not test with multiple values which is the purpose of the PR !

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthority.java [hunk 5]
```diff
@@ -28,10 +32,10 @@
 public @interface RequiresAuthority {
 
     /**
-     * @return The authority (as returned by {@link GrantedAuthority#getAuthority()} needed to execute method. Value is
+     * @return The authority list (as returned by {@link GrantedAuthority#getAuthority()} needed to execute method. Value is
      * ignored if {@link #authority()} is defined.
      */
-    String value() default StringUtils.EMPTY;
+    String[] value() default {};
```
Comment (sgandon) [inline daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthority.java]: Why the feature (multi ored authorities ) is only made available on *default* and not on *values* ?

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/47/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/47/)

File: daikon-spring/daikon-spring-security/src/test/java/org/talend/daikon/security/access/RequiresTestComponent.java (hunk 4 above)
Comment (frassinier) [reply daikon-spring/daikon-spring-security/src/test/java/org/talend/daikon/security/access/RequiresTestComponent.java]: Fixed

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/48/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/48/)

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthority.java (hunk 5 above)
Comment (frassinier) [reply daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthority.java]: I've fixed that

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/50/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/50/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/51/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/51/)

**Comment by sgandon** [comment]
thanks Fabien

**Comment by sgandon** [comment]
please merge (or let me know if you don't have the rights to do so)

**Pull request merged**